#!/usr/bin/env python
PACKAGE = "simple_camera_transformer"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("resize",                  double_t, 0, "Resize factor", 1, 0.1, 1)

gen.add("enable_color_correct",    bool_t,   0, "Enable Color Adjustment", False)
gen.add("cc_alpha",                double_t, 0, "Contrast", 1, 0, 10)
gen.add("cc_beta",                 double_t, 0, "Brightness", 0, -2550, 2550)

gen.add("enable_sharpen",          bool_t,   0, "Enable Sharpening", False)
gen.add("sharp_weight",            double_t, 0, "Sharpness", .5, -5, 5)
gen.add("sharp_kernel",            int_t,    0, "Sharpness Kernel Size", 1, 1, 10)

exit(gen.generate(PACKAGE, "simple_camera_transformer", "Transformer"))
